src/azure-cli-core/azure/cli/core/tests/test_arm.py

```diff
@@ -138,24 +129,81 @@ def test_resource_parse(self):
+                               'vaults/vault_name/backupFabrics/fabric_name/protectionContainers/container_name/'
+                               'protectedItems/item_name/recoveryPoint/recovery_point_guid',
+                'expected': {
+                    'subscription': '00000',
+                    'resource_group': 'myRg',
+                    'namespace': 'Microsoft.RecoveryServices',
+                    'type': 'vaults',
+                    'name': 'vault_name',
+                    'child_type_1': 'backupFabrics',
+                    'child_name_1': 'fabric_name',
+                    'child_type_2': 'protectionContainers',
+                    'child_name_2': 'container_name',
+                    'child_type_3': 'protectedItems',
+                    'child_name_3': 'item_name',
+                    'child_type_4': 'recoveryPoint',
+                    'child_name_4': 'recovery_point_guid',
+                    'resource_parent': 'vaults/vault_name/backupFabrics/fabric_name/'
+                                       'protectionContainers/container_name/protectedItems/item_name/',
+                    'resource_namespace': 'Microsoft.RecoveryServices',
+                    'resource_type': 'recoveryPoint',
+                    'resource_name': 'recovery_point_guid'
+                }
+            },
+            {
+                'resource_id': '/subscriptions/mySub/resourceGroups/myRg/providers/'
+                               'Microsoft.Provider1/resourceType1/name1/resourceType2/name2/providers/'
+                               'Microsoft.Provider3/resourceType3/name3',
+                'expected': {
+                    'subscription': 'mySub',
+                    'resource_group': 'myRg',
+                    'namespace': 'Microsoft.Provider1',
+                    'type': 'resourceType1',
+                    'name': 'name1',
+                    'child_namespace_1': None,
+                    'child_type_1': 'resourceType2',
+                    'child_name_1': 'name2',
+                    'child_namespace_2': 'Microsoft.Provider3',
+                    'child_type_2': 'resourceType3',
+                    'child_name_2': 'name3',
+                    'resource_parent': 'resourceType1/name1/resourceType2/name2/providers/Microsoft.Provider3/',
+                    'resource_namespace': 'Microsoft.Provider1',
+                    'resource_type': 'resourceType3',
+                    'resource_name': 'name3'
+                }
             }
         ]
         for test in tests:
+            self.assertTrue(is_valid_resource_id(test['resource_id']))
             kwargs = parse_resource_id(test['resource_id'])
             for key in test['expected']:
                 try:
                     self.assertEqual(kwargs[key], test['expected'][key])
                 except KeyError:
                     self.assertTrue(key not in kwargs and test['expected'][key] is None)
 
+        invalid_ids = [
```

Comment:
Also, please test a case for "resource_id" where things are insufficiently supplied. For example, child_name_1, child_type_2, child_namespace_3 or something.  Instead of just creating an ID with bizarre None values, it should raise a useful error. We've definitely run into bugs with this in the past.

Reply:
Done, please check again